internal/worker/controlsocket/worker.go

```diff
@@ -157,23 +215,30 @@ func (w *Worker) handleAddMetricsUser(resp http.ResponseWriter, req *http.Reques
 	w.writeResponse(resp, code, infof("created user %q", parsedBody.Username))
 }
 
-func (w *Worker) addMetricsUser(username, password string) (int, error) {
+func (w *Worker) addMetricsUser(ctx context.Context, username string, password auth.Password) (int, error) {
 	err := validateMetricsUsername(username)
 	if err != nil {
 		return http.StatusBadRequest, err
 	}
 
-	if password == "" {
-		return http.StatusBadRequest, errors.NotValidf("empty password")
+	creatorUser, err := w.userService.GetUserByName(ctx, userCreator)
+	if err != nil {
+		return http.StatusInternalServerError, errors.Annotatef(err, "retrieving creator user %q: %v", userCreator, err)
 	}
 
-	user, err := w.config.State.AddUser(username, username, password, userCreator)
+	_, _, err = w.userService.AddUser(ctx, service.AddUserArg{
```

Comment:
Would there be any benifit doing `w.catacomb.Context(ctx)` here (and in other service calls) allow the worker kill method to bring this down too?

Reply:
The server is tied to the catacomb already, so if the catacomb dies, the server dies, thus the request dies.